### PR TITLE
fix: `TransferTransaction.GetTokenTransfers()`

### LINKED ISSUE #320: Regression TransferTransaction.GetTokenTransfers returns nft transfers
https://github.com/hashgraph/hedera-sdk-go/blob/5eb38d215e6ec3a2f81111322eb7b51a238c668a/transfer_transaction.go#L104-L114

Should return fungible token transfers.

### PR DESCRIPTION
**Pull request by danielakhterov**
Signed-off-by: Daniel Akhterov <daniel@launchbadge.com>

**Description**:

**Related issue(s)**:

Fixes #320 

**Notes for reviewer**:

**Checklist**

- [X] Documented (Code comments, README, etc.)
- [X] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by xin-hedera** [comment]
had a hard time reading the token transfers stuff, refactored it here with the hope to make it easier to understand:

https://github.com/hashgraph/hedera-sdk-go/compare/fix/get-token-transfers...xin-hedera:fix/get-token-transfers?expand=1

ran the unit tests manually. still not confident since there isn't a good test coverage.

**Comment by danielakhterov** [comment]
@xin-hedera May I asked which part was difficult to read? Also, if you have recommendations on tests to add I can add to this PR I'm all ears.

**Comment by xin-hedera** [comment]
> @xin-hedera May I asked which part was difficult to read? Also, if you have recommendations on tests to add I can add to this PR I'm all ears.

not the changes which fix the bug

I see the previous changes to fix the inconsistent order of transfers. It uses a two-level index map nested data structure, I don't think that's easy to maintain / understand.

**Comment by danielakhterov** [comment]
Note that the current index maps will likely be changed back to what they used to be and a new implementation that simply adding sorts the transfers by account ID and token ID before converting into a protobuf list will be used instead. This would allow for users to add transfers out of order and still result with the same `TransferTransaction` even across SDKs.

**Review by andrix10 — DISMISSED** [review]

**Comment by xin-hedera** [comment]
> Note that the current index maps will likely be changed back to what they used to be and a new implementation that simply adding sorts the transfers by account ID and token ID before converting into a protobuf list will be used instead. This would allow for users to add transfers out of order and still result with the same `TransferTransaction` even across SDKs.

Fair point.

File: token_transfer_transaction_unit_test.go [hunk 1]
```diff
@@ -8,6 +8,50 @@ import (
...
+	assert.Equal(t, tokenTransfers[tokenID1][1], TokenTransfer{AccountID: accountID2, Amount: -10})
+	assert.Equal(t, len(tokenTransfers[tokenID2]), 2)
+	assert.Equal(t, tokenTransfers[tokenID2][0], TokenTransfer{AccountID: accountID1, Amount: 10})
+	assert.Equal(t, tokenTransfers[tokenID2][1], TokenTransfer{AccountID: accountID2, Amount: -10})
+
+	assert.Equal(t, len(nftTransfers), 2)
+	assert.Equal(t, len(nftTransfers[tokenID3]), 1)
+	assert.Equal(t, nftTransfers[tokenID3][0], TokenNftTransfer{SenderAccountID: accountID1, ReceiverAccountID: accountID2, SerialNumber: 9})
+	assert.Equal(t, len(nftTransfers[tokenID4]), 1)
+	assert.Equal(t, nftTransfers[tokenID4][0], TokenNftTransfer{SenderAccountID: accountID2, ReceiverAccountID: accountID1, SerialNumber: 10})
```
Comment (xin-hedera) [inline token_transfer_transaction_unit_test.go:52]: nit, can make the asserts more organized

```suggestion
	assert.Equal(t, hbarTransfers, map[AccountID]Hbar{
		accountID1: amount,
		accountID2: amount.Negated(),
	})

	assert.Equal(t, tokenTransfers, map[TokenID][]TokenTransfer{
		tokenID1: {
			{AccountID: accountID1, Amount: 10},
			{AccountID: accountID2, Amount: -10},
		},
		tokenID2: {
			{AccountID: accountID1, Amount: 10},
			{AccountID: accountID2, Amount: -10},
		},
	})

	assert.Equal(t, nftTransfers, map[TokenID][]TokenNftTransfer{
		tokenID3: {{SenderAccountID: accountID1, ReceiverAccountID: accountID2, SerialNumber: 9}},
		tokenID4: {{SenderAccountID: accountID2, ReceiverAccountID: accountID1, SerialNumber: 10}},
	})
```

**Review by xin-hedera — DISMISSED** [review]
just one nit

**Review by xin-hedera — APPROVED** [review]

**Pull request merged**